bucket/r42patched.json

```diff
@@ -0,0 +1,71 @@
+    "homepage": "https://www.r-project.org",
+    "license": "GPL-2.0-only",
+    "notes": [
+        "For source packages installation, please install 'rtools'.",
+        "",
+        "You'll need to type 'r.ps1' or 'r.cmd' to run R, because in Powershell 'r' runs the last command. Alternatively 'rterm' can be used to start the interactive R terminal session.",
+        "",
+        "You can remove Powershell's 'r' command with:",
+        "    rm alias:\\r",
+        "",
+        "... but this only affects your current session: if you'd like to remove it for all future sessions you need to add the command above to your Powershell profile.",
+        "",
+        "Annoying, right?! You might want to check out Pshazz (scoop install pshazz)--this has a plugin to remove some crazy aliases from Powershell, as well as many other improvements."
+    ],
+    "url": "https://cran.rstudio.com/bin/windows/base/R-4.2.0patched-win.exe",
+    "hash": "3E6C6046203A968985079A9256BCDD1AA7CF738C88F2AC3C4BE292BF80C7B43E",
+    "innosetup": true,
+    "architecture": {
+        "64bit": {
+            "pre_install": "Rename-Item \"$dir\\bin\\R,2.exe\" 'R.exe'",
+            "bin": [
+                "bin\\x64\\R.exe",
+                "bin\\x64\\Rcmd.exe",
+                "bin\\x64\\Rgui.exe",
+                "bin\\x64\\Rscript.exe",
+                "bin\\x64\\Rterm.exe"
+            ],
+            "shortcuts": [
+                [
+                    "bin\\x64\\Rgui.exe",
+                    "R 4.2.0-patched"
+                ]
+            ]
+        }
+    },
+    "installer": {
+        "script": [
+            "Remove-Item \"$dir\\bin\\R,*.exe\" -Force",
+            "Copy-Item \"$dir\\bin\\R.exe\" \"$dir\\bin\\Rscript.exe\"",
+            "if (Test-Path (appdir 'rtools' $global)) {",
+            "   $renvloc = \"$dir\\etc\\Renviron.site\"",
+            "   $rtoolsloc = versiondir 'rtools' (current_version 'rtools' $global) $global",
+            "   if (Test-Path $renvloc) {",
+            "       $renv = Get-Content $renvloc -Encoding ASCII",
+            "   } else {",
+            "       $renv = @()",
+            "   }",
+            "   $rtoolspath = \"$rtoolsloc\\usr\\bin\"",
+            "   if ($renv -cmatch 'PATH=') {",
+            "       $renv = $renv -replace 'PATH=\"(.*)', ('PATH=\"' + $rtoolspath + ';$1')",
+            "   } else {",
+            "       $renv += \"`r`nPATH=\", '\"', $rtoolspath, ';${PATH}\"' -join ''",
+            "   }",
+            "   $renv | Set-Content $renvloc -Encoding ASCII",
+            "}"
+        ]
+    },
+    "checkver": {
+        "url": "https://cloud.r-project.org/bin/windows/contrib/r-release/",
+        "regex": "R-([\\d.]+)"
+    },
+    "autoupdate": {
+        "url": "https://cloud.r-project.org/bin/windows/base/R-$version-win.exe",
+        "hash": {
+            "url": "$baseurl/md5sum.R-$version.txt"
+        }
+    }
```

Comment:
```suggestion
    "checkver": {
        "url": "https://cran.rstudio.com/bin/windows/base/rpatched.html",
        "regex": "R-([\\d.]+)",
        "replace": "${1}patched"
    },
    "autoupdate": {
        "url": "https://cloud.r-project.org/bin/windows/base/R-$version-win.exe",
        "hash": {
            "url": "$baseurl/md5sum.R-$version.txt"
        }
    }
```

Reply:
I am not sure if this package needs *autoupdate*, but you need to use **replace** so that *checkver* outputs the version as `4.2.0patched`.